### ISSUE TITLE
select! macro produces clippy warning

### ISSUE THREAD

**Issue by idubrov** [rust-lang/futures-rs#2009]
The following code ([Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=63fcf22c7afe03190e5daf029bc43fae)):

```rust
async fn hello() {
    let mut future = futures::future::ready(42);
    futures::select! {
        _ = future => {}
    }
}
```

produces clippy warning: 

```
warning: The function/method `::futures::async_await::assert_fused_future` doesn't need a mutable reference
 --> src/lib.rs:4:13
  |
4 |         _ = future => {}
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::unnecessary_mut_passed)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_mut_passed

warning: The function/method `::futures::async_await::assert_unpin` doesn't need a mutable reference
 --> src/lib.rs:4:13
  |
4 |         _ = future => {}
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_mut_passed
```

**Comment by dunnock** [comment]
Actually same issue even when using [example from the official doc](https://rust-lang.github.io/async-book/06_multiple_futures/03_select.html):
```
warning: The function/method `::futures::async_await::assert_fused_future` doesn't need a mutable reference
 --> src/main.rs:19:14
  |
4 | |     future::FutureExt, // for `.fuse()`
  | |____________________________________________________^
...
19|           () = t1 => println!("task one completed first"),
  |  ______________^
  |
  = note: `#[warn(clippy::unnecessary_mut_passed)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_mut_passed

warning: The function/method `::futures::async_await::assert_unpin` doesn't need a mutable reference
 --> src/main.rs:19:14
  |
4 | |     future::FutureExt, // for `.fuse()`
  | |____________________________________________________^
...
19|           () = t1 => println!("task one completed first"),
  |  ______________^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_mut_passed
```

**Comment by taiki-e** [comment]
Fixed in #1981

**Comment by dbrgn** [comment]
Can this be closed, now that #1981 is merged?

**Comment by Nemo157** [comment]
Yep, running Clippy on that playground no longer emits the warning, thanks.
